docs/user-guide/reduction-workflow-guidelines.md

```diff
@@ -0,0 +1,139 @@
+
+### S.3: Avoid dependencies of output metadata on large data
+
+**Reason**
+Adding dependencies on large data to the output metadata extraction may lead to large data being kept alive in memory.
+
+**Note**
+Most of this is avoid by following S.2.
+A bad exampleAvoid would be to write to total raw counts to the output metadata, as this would require keeping the large data alive in memory, unless it is ensured that the task runs early.
+
+### S.4: Handle single- and double-precision transparently
+
+**Reason**
+Single-precision may be sufficient for most data.
+By writing workflows transparently for single- and double-precision, we avoid future changes if we either want to use single-precision for performance reasons or double-precision for accuracy reasons.
+
+**Note**
+This affects coordinates and data values independently.
+- If input counts are single-precision, the reduced intensity should be single-precision, and equivalently for double-precision.
+- If input coordinates are single-precision, derived coordinates should be single-precision, and equivalently for double-precision.
+
+**Note**
+This will allow for changing the precision of the entire workflow by choosing a precision when loading the input data.
+
+**Example**
+- If time-of-flight is single-precision, wavelength and momentum transfer should be single-precision.
+- In counts are single-precision, reduced intensity should be single-precision.
+
+### S.5: Switches to double-precision shall be deliberate, explicit, and documented
+
+**Reason**
+Some workflows may require switching to double-precision at a certain point in the workflow.
+This should be a deliberate choice, and the reason for the switch should be documented.
+
+### S.6: Propagation of uncertainties in broadcast operations should support "drop" and "upper-bound" strategies, "upper-bound" shall be the default
+
+**Reason**
+Unless explicitly computed, the exact propagation of uncertainties in broadcast operations is not well-defined.
+Dropping uncertainties is not desireable in general, as it may lead to underestimation of the uncertainties, but we realize that the upper-bound approach may not be suitable in all cases.
+We should therefore support two strategies, "drop" and "upper-bound", and "upper-bound" should be the default.
+
+**Note**
+See [Systematic underestimation of uncertainties by widespread neutron-scattering data-reduction software](http://dx.doi.org/10.3233/JNR-220049) for a discussion of the topic.
+TODO Add reference to upper-bound approach.
+
+### S.7: Do not write files or make write requests to services such as SciCat in providers
+
+**Reason**
+Providers should be side-effect free, and should not write files or make write requests to services.
+
+**Note**
+Workflow may run many times, or in parallel, or task may be retried after failure, and we want to avoid side-effects in these cases.
```

Comment:
```suggestion
Workflows may run many times, or in parallel, or tasks may be retried after failure, and we want to avoid side-effects in these cases.
```